ote_cli/ote_cli/tools/demo.py

```diff
@@ -0,0 +1,202 @@
+from ote_cli.utils.config import override_parameters
+from ote_cli.utils.importing import get_impl_class
+from ote_cli.utils.loading import load_model_weights
+from ote_cli.utils.parser import (
+    add_hyper_parameters_sub_parser,
+    gen_params_dict_from_args,
+)
+
+ESC_BUTTON = 27
+
+
+def parse_args():
+    """
+    Parses command line arguments.
+    """
+
+    pre_parser = argparse.ArgumentParser(add_help=False)
+    pre_parser.add_argument("template")
+    parsed, _ = pre_parser.parse_known_args()
+    # Load template.yaml file.
+    template = find_and_parse_model_template(parsed.template)
+    # Get hyper parameters schema.
+    hyper_parameters = template.hyper_parameters.data
+    assert hyper_parameters
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument("template")
+    parser.add_argument(
+        "-i",
+        "--input",
+        required=True,
+        help="Source of input data: images folder, image, webcam.",
+    )
+    parser.add_argument(
+        "--load-weights",
+        required=True,
+        help="Load only weights from previously saved checkpoint",
+    )
+    parser.add_argument(
+        "--fit-to-size",
+        nargs=2,
+        type=int,
+        help="Width and Height space-separated values. "
+        "Fits displayed images to window with specified Width and Height. "
+        "This options applies to result visualisation only.",
+    )
+    parser.add_argument(
+        "--loop", action="store_true", help="Enable reading the input in a loop."
+    )
+    parser.add_argument(
+        "--delay", type=int, default=3, help="Frame visualization time in ms."
+    )
+    parser.add_argument(
+        "--labels", nargs="+", required=True, help="A space-separated labels list."
+    )
+    parser.add_argument(
+        "--display-perf",
+        action="store_true",
+        help="This option enables writing performance metrics on displayed frame. "
+        "These metrics take into account not only model inference time, but also "
+        "frame reading, pre-processing and post-processing.",
+    )
+
+    add_hyper_parameters_sub_parser(parser, hyper_parameters, modes=("INFERENCE",))
+
+    return parser.parse_args(), template, hyper_parameters
+
+
+def get_predictions(task, frame):
+    """
+    Returns list of predictions made by task on frame and time spent on doing prediction.
+    """
+
+    empty_annotation = AnnotationSceneEntity(
+        annotations=[], kind=AnnotationSceneKind.PREDICTION
+    )
+
+    item = DatasetItemEntity(
+        media=Image(frame),
+        annotation_scene=empty_annotation,
+    )
+
+    dataset = DatasetEntity(items=[item])
+
+    start_time = time.perf_counter()
+    predicted_validation_dataset = task.infer(
+        dataset,
+        InferenceParameters(is_evaluation=True),
+    )
+    elapsed_time = time.perf_counter() - start_time
+    item = predicted_validation_dataset[0]
+    return item.annotation_scene.annotations, elapsed_time
```

Comment:
It's better use `item.get_annotations()` rather than get those directly from the annotation scene. I guess the result will be the same in this case (all ROIs are covering the whole frame), but in general the recommendation is to do so.

Reply:
fixed